ras/lshw.py

```diff
@@ -210,14 +210,18 @@ def test_lshw_options(self):
         """
         self.log.info("===============Verifying -disable, -enable, -quiet,"
                       " and -numeric options ==============")
-        if "interface" in self.run_cmd_out("lshw -disable network| "
-                                           "grep -i interface"):
+        if "Ethernet interface" in self.run_cmd_out("lshw -disable network| "
+                                                    "grep -i \"ethernet "
+                                                    "interface\""):
             self.is_fail += 1
-            self.fail_cmd.append("lshw -disable network|grep -i interface")
-        if "interface" not in self.run_cmd_out("lshw -enable network| "
-                                               "grep -i interface"):
+            self.fail_cmd.append(
+                "lshw -disable network|grep -i \"ethernet interface\"")
+        if "Ethernet interface" not in self.run_cmd_out("lshw -enable network| "
+                                                        "grep -i \"ethernet " 
+                                                        "interface\""):
```

Comment:
the command fails on both powevm and powernv

lshw -enable network| ""grep -i \"ethernet " "interface\"
grep:  interface": No such file or directory
$ echo $?
2

Reply:
a better rule is to not use shell pipe in python code, how about

cmd = "lshw -enable network"
for res in self.run_cmd_out(cmd).split():
    if "Ethernet interface" not in:

Reply:
it is working as expected Abdul
#  lshw -disable network| grep  -i "ethernet interface"
#  lshw -enable network| grep  -i "ethernet interface"
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
#